Sources/Swift/Helper/SentrySDK.swift

```diff
@@ -361,12 +361,24 @@ import Foundation
     /// - parameter timeout: The time to wait for the SDK to complete the flush.
     @objc(flush:)
     public static func flush(timeout: TimeInterval) {
-        SentrySDKInternal.flush(timeout: timeout)
+        // Logger gets max 25% of the timeout
+        let loggerTimeout = timeout * 0.25 
+        
+        // Calculate how long logger flush took
+        let startTime = CFAbsoluteTimeGetCurrent()
+        flushLogger(timeout: loggerTimeout)
+        let endTime = CFAbsoluteTimeGetCurrent()
+        let loggerElapsedTime = endTime - startTime
+        
+        // Main flush gets the remaining time
+        let remainingTime = max(timeout * 0.75, timeout - loggerElapsedTime)
```

Comment:
`m`: I think the log API should offer a method for storing the logs in memory to disk and then we call the `SentrySDKInternal.flush(timeout: remainingTime)`. With that approach we only have one single flush implementation in the HTTPTransport and not in the batcher and in the HTTPTransport.